src/mlpack/methods/bayesian_linear_regression/bayesian_linear_regression.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * @file bayesian_linear_regression.cpp
+ * @author Clement Mercier 
+ *
+ * Implementation of Bayesian linear regression.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "bayesian_linear_regression.hpp"
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/core/util/timers.hpp>
+
+using namespace mlpack;
+using namespace mlpack::regression;
+
+BayesianLinearRegression::BayesianLinearRegression(const bool centerData,
+                                                   const bool scaleData,
+                                                   const int nIterMax,
+                                                   const double tol) :
+  centerData(centerData),
+  scaleData(scaleData),
+  nIterMax(nIterMax),
+  tol(tol)
+{/* Nothing to do */}
+
+double BayesianLinearRegression::Train(const arma::mat& data,
+                                       const arma::rowvec& responses)
+{
+  Timer::Start("bayesian_linear_regression");
+
+  arma::mat phi;
+  arma::rowvec t;
+  arma::colvec eigVal;
+  arma::mat eigVec;
+
+  // Preprocess the data. Center and scale.
+  responsesOffset = CenterScaleData(data,
+                                    responses,
+                                    phi,
+                                    t);
+
+  if (!arma::eig_sym(eigVal, eigVec, arma::symmatu(phi * phi.t())))
+  {
+    Log::Fatal << "BayesianLinearRegression::Train(): Eigendecomposition "
+               << "of covariance failed!";
```

Comment:
```suggestion
               << "of covariance failed!" << std::endl;
```
`Log::Fatal` needs a `std::endl` in order to throw an exception. :)